java/ql/lib/ext/java.lang.model.yml

```diff
@@ -34,16 +34,21 @@ extensions:
       - ["java.lang", "AbstractStringBuilder", True, "toString", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "Appendable", True, "append", "", "", "Argument[-1]", "ReturnValue", "value", "manual"]
       - ["java.lang", "Appendable", True, "append", "", "", "Argument[0]", "Argument[-1]", "taint", "manual"]
+      - ["java.lang", "AssertionError", False, "AssertionError", "(Object)", "", "Argument[0]", "Argument[-1].SyntheticField[java.lang.Throwable.message]", "value", "manual"]
       - ["java.lang", "CharSequence", True, "charAt", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "CharSequence", True, "subSequence", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
       - ["java.lang", "CharSequence", True, "toString", "", "", "Argument[-1]", "ReturnValue", "taint", "manual"]
+      - ["java.lang", "Class", False, "cast", "(Object)", "", "Argument[0]", "ReturnValue", "taint", "manual"]
```

Comment:
I changed taint to value in https://github.com/github/codeql/pull/11968/commits/f5b98f984f4856c59b7a8b8aec446015e2273a6b for now.

Dumb question: how would I properly add this to `CastingExpr`? (I assume you mean [this CastingExpr](https://github.com/github/codeql/blob/604d5f0c713713d673c392262dd3312e96ad1d17/java/ql/lib/semmle/code/java/Expr.qll#L1121)?)

Reply:
> (I assume you mean [this CastingExpr](https://github.com/github/codeql/blob/604d5f0c713713d673c392262dd3312e96ad1d17/java/ql/lib/semmle/code/java/Expr.qll#L1121)?)

Yes.

> how would I properly add this to `CastingExpr`?

I was imagining that it'd be something like:

```ql
class CastingExpr extends Expr {
  CastingExpr() {
    this instanceof @castexpr or
    this instanceof @safecastexpr or
    this instanceof @implicitcastexpr or
    this instanceof @implicitnotnullexpr or
    this instanceof @implicitcoerciontounitexpr or
    this instanceof @unsafecoerceexpr or
    this.(MethodAccess).getMethod().hasQualifiedName("java.lang", "Class", "cast")
  }

  /** Gets the target type of this casting expression. */
  Expr getTypeExpr() { 
    result.isNthChildOf(this, 0) or
    result = this.(GenericCall).getATypeArgument(this.(GenericCall).getCallee().getATypeParameter()) // ????
  }

  /** Gets the expression to which the casting operator is applied. */
  Expr getExpr() { result.isNthChildOf(this, 1) or result = this.(MethodAccess).getArgument(0) }
```

But this probably has deeper implications than what it could initially seem, so happy to leave this as MaD for now.

Reply:
Ah, thanks for the demo, that makes sense.

> But this probably has deeper implications than what it could initially seem, so happy to leave this as MaD for now.
Sounds good, I'll leave as is.